torch/distributed/algorithms/ddp_comm_hooks/powerSGD_hook.py

```diff
@@ -268,7 +268,8 @@ def decompress(fut):
 
         for p, q, tensor in zip(ps, qs, high_rank_tensors):
             torch.matmul(p, q.t(), out=tensor)
-            assert not torch.any(torch.isnan(tensor))
+        if torch.cuda.is_available():
```

Comment:
The hook isn't supported for CPU use cases currently right? Also, we might have the case where CUDA is available but user is using Gloo backend but with CPU-only tensors. Would it be better to check if `device` is a cuda device here?

Reply:
Ideally the explicit synchronization should not be needed here, since there is already a `wait()` in every precursor future. There is a potential [bug](https://github.com/pytorch/pytorch/issues/49477
) in the future library now, and it is only supported on NCCL backend at this time.

Prefer no change here, and will update it accordingly once the potential bug in the future library is resolved.